tests/auth.rs

```diff
@@ -0,0 +1,361 @@
+        .returning(None)
+        .expect_get_property(Some(vec!["request", "host"]))
+        .returning(Some("cars.toystore.com".as_bytes()))
+        .expect_get_property(Some(vec!["request", "method"]))
+        .returning(Some("GET".as_bytes()))
+        .expect_get_property(Some(vec!["request", "scheme"]))
+        .returning(Some("http".as_bytes()))
+        .expect_get_property(Some(vec!["request", "path"]))
+        .returning(Some("/admin/toy".as_bytes()))
+        .expect_get_property(Some(vec!["request", "protocol"]))
+        .returning(Some("HTTP".as_bytes()))
+        .expect_get_property(Some(vec!["request", "time"]))
+        .returning(None)
+        .expect_get_property(Some(vec!["destination", "address"]))
+        .returning(Some("127.0.0.1:8000".as_bytes()))
+        .expect_get_property(Some(vec!["destination", "port"]))
+        .returning(Some("8000".as_bytes()))
+        .expect_get_property(Some(vec!["source", "address"]))
+        .returning(Some("127.0.0.1:45000".as_bytes()))
+        .expect_get_property(Some(vec!["source", "port"]))
+        .returning(Some("45000".as_bytes()))
+        // retrieving tracing headers
+        .expect_get_header_map_value(Some(MapType::HttpRequestHeaders), Some("traceparent"))
+        .returning(None)
+        .expect_get_header_map_value(Some(MapType::HttpRequestHeaders), Some("tracestate"))
+        .returning(None)
+        .expect_get_header_map_value(Some(MapType::HttpRequestHeaders), Some("baggage"))
+        .returning(None)
+        .expect_grpc_call(
+            Some("authorino-cluster"),
+            Some("envoy.service.auth.v3.Authorization"),
+            Some("Check"),
+            Some(&[0, 0, 0, 0]),
+            Some(&[
+                10, 217, 1, 10, 23, 10, 21, 10, 19, 18, 15, 49, 50, 55, 46, 48, 46, 48, 46, 49, 58,
+                52, 53, 48, 48, 48, 24, 0, 18, 22, 10, 20, 10, 18, 18, 14, 49, 50, 55, 46, 48, 46,
+                48, 46, 49, 58, 56, 48, 48, 48, 24, 0, 34, 141, 1, 10, 0, 18, 136, 1, 18, 3, 71,
+                69, 84, 26, 30, 10, 10, 58, 97, 117, 116, 104, 111, 114, 105, 116, 121, 18, 16, 97,
+                98, 105, 95, 116, 101, 115, 116, 95, 104, 97, 114, 110, 101, 115, 115, 26, 14, 10,
+                7, 58, 109, 101, 116, 104, 111, 100, 18, 3, 71, 69, 84, 26, 38, 10, 5, 58, 112, 97,
+                116, 104, 18, 29, 47, 100, 101, 102, 97, 117, 108, 116, 47, 114, 101, 113, 117,
+                101, 115, 116, 47, 104, 101, 97, 100, 101, 114, 115, 47, 112, 97, 116, 104, 34, 10,
+                47, 97, 100, 109, 105, 110, 47, 116, 111, 121, 42, 17, 99, 97, 114, 115, 46, 116,
+                111, 121, 115, 116, 111, 114, 101, 46, 99, 111, 109, 50, 4, 104, 116, 116, 112, 82,
+                4, 72, 84, 84, 80, 82, 20, 10, 4, 104, 111, 115, 116, 18, 12, 97, 117, 116, 104,
+                99, 111, 110, 102, 105, 103, 45, 65, 90, 0,
+            ]),
+            Some(5000),
```

Comment:
We'll need to set the timeout config when https://github.com/Kuadrant/wasm-shim/pull/101 is merged